Project.toml

```diff
@@ -1,7 +1,7 @@
 name = "ParticleFilters"
 uuid = "c8b314e2-9260-5cf8-ae76-3be7461ca6d0"
 repo = "https://github.com/JuliaPOMDP/ParticleFilters.jl"
-version = "0.5.1"
+version = "0.5.2"
```

Comment:
I think this should be 0.6.0 now since it breaks existing code.

Reply:
According to SemVer this would be okay since [all major version 0 releases may be incompatible with eachother](https://semver.org/). But in Julia there is the [convention](https://julialang.github.io/Pkg.jl/stable/compatibility/#Caret-specifiers-1):

> While the semver specification says that all versions with a major version of 0 are incompatible with each other, we have made that choice that a version given as 0.a.b is considered compatible with 0.a.c if a != 0 and c >= b.

Reply:
Yeah, I actually looked this up (https://semver.org/#how-should-i-handle-deprecating-functionality), and I am a bit confused about it. It looks like what you're actually supposed to do is deprecate with a minor version release, and then remove in the next major version (so in this case deprecate in 0.5.2 and remove in 0.6.0).

Before julia 1.5, deprecation warnings were on by default, so there was a big performance penalty for deprecating things, but now they are off by default except in tests. So, I think 0.5.2 is actually correct.